internal/cli/server_upgrade.go

```diff
@@ -389,10 +389,42 @@ func (c *ServerUpgradeCommand) upgradeRunner(
 	s.Update("Previous runner uninstalled")
 	s.Done()
 
-	// TODO(mitchellh): This creates a new auth token for the new runner.
-	// In the future, we need to invalidate the old token. We don't have
-	// the functionality to do this today.
-	return installRunner(ctx, installOpts.Log, client, c.ui, p, advertiseAddr)
+	if odc, ok := p.(serverinstall.OnDemandRunnerConfigProvider); ok {
+		odr := odc.OnDemandRunnerConfig()
+
+		// We attempt to look up the default runner profile from the previous
+		// installation. If we find it, we get the ID, so we can delete it after
+		// the new runner profile is set up
+		oldRunnerConfig, err := client.GetOnDemandRunnerConfig(ctx, &pb.GetOnDemandRunnerConfigRequest{
+			Config: &pb.Ref_OnDemandRunnerConfig{
+				Id:   odr.Id,
+				Name: odr.Name,
+			}})
+		if err != nil && status.Code(err) != codes.NotFound {
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		} else if err != nil && status.Code(err) == codes.NotFound {
+			c.ui.Output("Waypoint default runner profile not found, creating new profile", terminal.WithWarningStyle())
+		} else {
+			odr = &pb.OnDemandRunnerConfig{
+				Id:                   oldRunnerConfig.Config.Id,
+				Name:                 oldRunnerConfig.Config.Name,
+				TargetRunner:         oldRunnerConfig.Config.TargetRunner,
+				OciUrl:               "hashicorp/waypoint:latest",
```

Comment:
This image URL can come from a CLI flag too, so we should use that instead of the static string here in case they provide their own image URL